### PR TITLE
[Windows] Add shell symlinks

### PR DESCRIPTION
**Pull request by al-cheb**
# Description
Add shell symlinks for Windows:

| Name         | Target                            |
| ------------ | --------------------------------- |
| gitbash.exe  | C:\Program Files\Git\bin\bash.exe |
| msys2bash.cmd | C:\msys64\usr\bin\bash.exe        |
| wslbash.exe  | C:\windows\System32\bash.exe      |

#### Related issue:
https://github.com/actions/virtual-environments/issues/1525

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Comment by al-cheb** [comment]
@eine, Could you please review this PR?

**Review by Ms230491 — APPROVED** [review]

**Comment by eine** [comment]
@al-cheb, I believe that `C:\msys64\usr\bin\bash.exe` and `C:\msys64\usr\bin\sh.exe` are, in fact, the same shell. I don't think it's necessary to add both of them. @lazka, wdyt?

At the same time, I would suggest adding a wrapper, instead of symlinking directly. Based on https://github.com/msys2/setup-msys2/blob/master/main.js#L165-L169, the wrapper might be `msys2bash.cmd`:

```
@echo off
setloca
IF NOT DEFINED MSYS2_PATH_TYPE set MSYS2_PATH_TYPE=strict
set CHERE_INVOKING=1
C:\msys64\usr\bin\bash.exe -leo pipefail %*
```

**Comment by al-cheb** [comment]
> @al-cheb, I believe that `C:\msys64\usr\bin\bash.exe` and `C:\msys64\usr\bin\sh.exe` are, in fact, the same shell. I don't think it's necessary to add both of them. @lazka, wdyt?
> 
> At the same time, I would suggest adding a wrapper, instead of symlinking directly. Based on https://github.com/msys2/setup-msys2/blob/master/main.js#L165-L169, the wrapper might be `msys2bash.cmd`:
> 
> ```
> @echo off
> setlocal
> IF NOT DEFINED MSYS2_PATH_TYPE set MSYS2_PATH_TYPE=strict
> IF NOT DEFINED MSYSTEM set MSYSTEM=mingw64
> set CHERE_INVOKING=1
> C:\msys64\usr\bin\bash.exe -leo pipefail %*
> ```

Thank you. Remove the link for `sh` shell and add a wrapper for bash.

File: images/win/scripts/Tests/Shell.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Describe "Shell" {
+    $IsWin16 = Test-IsWin16
+
+    $shellTestCases = @(
```
Comment (maxim-lobanov) [inline images/win/scripts/Tests/Shell.Tests.ps1:4]: Any reason why we can't add all 3 shells to this array and test?
I think `wslbash` can be added withoout any problem

Comment (maxim-lobanov) [reply images/win/scripts/Tests/Shell.Tests.ps1:4]:
```
$shellTestCases = ()
if (Test-IsWin16) {
  $shellTestCases += @{}
}
```

Comment (al-cheb) [reply images/win/scripts/Tests/Shell.Tests.ps1:2]: Thank you. Fixed.

**Review by maxim-lobanov — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Pull request merged**